test_data/asm/book/operations_and_links.asm

```diff
@@ -0,0 +1,37 @@
+// ANCHOR: links
+machine Main(latch, operation_id) {
+    Arith adder;
+
+    operation main<0> x, y -> z;
+
+    // on every row, add in the Adder
+    link 1 x, y -> z = adder.add;
```

Comment:
Was this 1 automatically generated? What's the asm that got compiled to this?

Reply:
No. This is a user-defined link, see book.

Reply:
I added more details in this source file.